File: Year-2023/LeetCode-January-Challenge/MinimumRoundstoCompleteAllTasks.js
var minimumRounds = function(tasks) {
    let res = 0, mp = {};
    for(let x of tasks) mp[x] ? mp[x]++ : mp[x] = 1;
    for(let x of Object.values(mp)){
        if(x <= 1) return -1;
        else if(x === 2) res++;
        else res += Math.ceil((x)/3.0);
    }
    return res;
};
let tasks = [2,2,3,3,2,4,4,4,4,4];
console.log(minimumRounds(tasks));

File: Year-2023/LeetCode-January-Challenge/MinimumRoundstoCompleteAllTasks.java
import java.util.*;
class Solution {
    public int minimumRounds(int[] tasks) {
        int res = 0;
        HashMap<Integer, Integer> mp = new HashMap<>();
        for(int x: tasks) mp.put(x, mp.getOrDefault(x, 0) + 1);
        for(Map.Entry<Integer, Integer> x:  mp.entrySet()){
            if(x.getValue() <= 1) return -1;
            else if(x.getValue() == 2) res++;
            else res += (Math.ceil((double)x.getValue()/3.0));
        }
        return res;
    }
}
public class MinimumRoundstoCompleteAllTasks{
    public static void main(String[] args) {
        int[] tasks = {2,2,3,3,2,4,4,4,4,4};
        Solution s = new Solution();
        System.out.println(s.minimumRounds(tasks));
    }
}